server/src/main/resources/broker.xml

```diff
@@ -19,16 +19,88 @@
+
+            An example of a login.config configured for Candlepin's embedded broker to allow both
+            external SSL connections and Candlepin's own in-vm connection is given below [2].
+
+            Beyond this, the security settings will need to be configured to allow the various users
+            to actually use the broker. Again, when using the embedded broker, Candlepin's in-vm
+            connection should be configured to perform any action to prevent operation failure. The
+            security settings section below has a commented-out block for the in-vm role to perform
+            any operation on any address/queue.
+
+            [1] candlepin.conf properties renaming:
+                - "candlepin.messaging.activemq.embedded.jaas_properties_login_name"
+                - "candlepin.messaging.activemq.embedded.jaas_certificate_login_name"
+
+            [2] Example login.config for an embedded broker with external clients:
+                CertificateLogin {
+                    org.apache.activemq.artemis.spi.core.security.jaas.TextFileCertificateLoginModule required
+                        debug=true
+                        org.apache.activemq.jaas.textfiledn.user="cert-users.properties"
+                        org.apache.activemq.jaas.textfiledn.role="cert-roles.properties";
+                };
+
+                PropertiesLogin {
+                    org.apache.activemq.artemis.spi.core.security.jaas.GuestLoginModule required
+                        debug=true
+                        credentialsInvalidate=true
+                        org.apache.activemq.jaas.guest.user="invm-user"
+                        org.apache.activemq.jaas.guest.role="invm-role";
+                };
+        -->
 
         <acceptors>
             <acceptor name="in-vm">vm://0</acceptor>
-            <!-- <acceptor name="netty">tcp://localhost:61617?sslEnabled=true;keyStorePath=${artemis.instance}/certs/artemis-server.ks;keyStorePassword=securepassword;needClientAuth=true;trustStorePath=${artemis.instance}/certs/artemis-server.ts;trustStorePassword=securepassword</acceptor> -->
+
+            <!-- <acceptor name="netty">
+                tcp://localhost:61613?protocols=STOMP;needClientAuth=true;sslEnabled=true;keyStorePath=${artemis.instance}/certs/server-side-keystore.jks;keyStorePassword=changeme;trustStorePath=${artemis.instance}/certs/server-side-truststore.jks;trustStorePassword=changeme
+            </acceptor> -->
         </acceptors>
 
+        <!-- set this to true if using external clients -->
         <security-enabled>false</security-enabled>
 
+        <security-settings>
+            <!--
+                Enable this when using the embedded broker with the security manager enabled.
+                Remember sure to update the role name to match the role given to the "guest"
+                in-vm user, following the login details provided above.
+
+            <security-setting match="#">
```

Comment:
I have a suspicion that these permissions are broader than what's actually needed. I experimented with this, and it seems to work but it's likely that i'm not testing all the execution paths:

```
                <security-setting match="job.jobs">
                        <permission type="createDurableQueue" roles="invm-role"/>
                        <permission type="deleteDurableQueue" roles="invm-role"/>
                        <permission type="send" roles="invm-role"/>
                        <permission type="consume" roles="invm-role"/>
                </security-setting>
                <security-setting match="event.default.#">
                        <permission type="createDurableQueue" roles="invm-role"/>
                        <permission type="deleteDurableQueue" roles="invm-role"/>
                        <permission type="send" roles="invm-role"/>
                        <permission type="consume" roles="invm-role"/>
                </security-setting>
```

Reply:
It's definitely a sledgehammer, but I don't want to limit or break in-vm functionality every time we add something going forward. I'm pretty sure we need the manage side of things today, and if not, it'll be coming soon with the rework of the status monitor.

I'm somewhat torn here in that I understand the concern for security, but I also feel that the connection Candlepin makes to its own embedded broker shouldn't be encumbered at all. That it even is subject to the security manager seems somewhat silly.

In any event, what is the update flow look like from your end if we have to update the permissions on a release?

Reply:
That's fair to say w.r.t limiting or breaking functionality. Something like that could go easily missed on our side, too. We'd need to update our installer where we will have the broker.xml puppetized. Let's leave it as you have it.